controllers/agentmachine_controller.go

```diff
@@ -425,40 +423,61 @@ func isValidAgent(agent *aiv1beta1.Agent, agentMachine *capiproviderv1alpha1.Age
+	} else {
+		conditions.MarkTrue(agentMachine, capiproviderv1alpha1.AgentReservedCondition)
 
-	for _, condition := range agent.Status.Conditions {
-		if condition.Type == aiv1beta1.InstalledCondition {
-			if condition.Status == "True" {
-				log.Info("Updating agentMachine status to Ready=true")
-				agentMachine.Status.Ready = true
-			} else if condition.Status == "False" {
-				if condition.Reason == aiv1beta1.InstallationFailedReason {
-					agentMachine.Status.FailureReason = (*clustererrors.MachineStatusError)(&condition.Reason)
-					message := condition.Message
-					agentMachine.Status.FailureMessage = &message
-				}
+		agent := &aiv1beta1.Agent{}
+		agentRef := types.NamespacedName{Name: agentMachine.Status.AgentRef.Name, Namespace: agentMachine.Status.AgentRef.Namespace}
+		if err := r.AgentClient.Get(ctx, agentRef, agent); err != nil {
+			log.WithError(err).Errorf("Failed to get agent %s", agentRef)
+			return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, err
+		}
+
+		agentInstalledCondition := openshiftconditionsv1.FindStatusCondition(agent.Status.Conditions, aiv1beta1.InstalledCondition)
+		if agentInstalledCondition.Status == "True" {
+			log.Info("Updating agentMachine NodeInstalledCondition to true")
+			conditions.MarkTrue(agentMachine, capiproviderv1alpha1.NodeInstalledCondition)
+		} else if agentInstalledCondition.Status == "False" {
+			if agentInstalledCondition.Reason == aiv1beta1.InstallationFailedReason {
+				conditions.MarkFalse(agentMachine, capiproviderv1alpha1.NodeInstalledCondition, capiproviderv1alpha1.NodeInstallationFailedReason,
+					clusterv1.ConditionSeverityError, agentInstalledCondition.Message)
+			} else {
+				conditions.MarkFalse(agentMachine, capiproviderv1alpha1.NodeInstalledCondition, capiproviderv1alpha1.NodeInstallationInProgressReason,
+					clusterv1.ConditionSeverityInfo, agentInstalledCondition.Message)
+				installationInProgress = true
 			}
-			break
 		}
 	}
 
+	conditions.SetSummary(agentMachine,
+		conditions.WithConditions(capiproviderv1alpha1.AgentReservedCondition, capiproviderv1alpha1.NodeInstalledCondition),
+		conditions.WithStepCounterIf(agentMachine.ObjectMeta.DeletionTimestamp.IsZero()),
+		conditions.WithStepCounter())
+
+	agentMachine.Status.Ready, _ = strconv.ParseBool(string(conditions.Get(agentMachine, clusterv1.ReadyCondition).Status))
+
 	if updateErr := r.Status().Update(ctx, agentMachine); updateErr != nil {
 		log.WithError(updateErr).Error("failed to update AgentMachine Status")
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, nil
 	}
 	if agentMachine.Status.Ready {
 		// No need to requeue in case the agentMachine is ready
 		return ctrl.Result{}, nil
 	}
-	return ctrl.Result{RequeueAfter: defaultRequeueWaitingForAgentToBeInstalled}, nil
+	if installationInProgress && origErr == nil {
+		return ctrl.Result{RequeueAfter: defaultRequeueWaitingForAgentToBeInstalled}, nil
```

Comment:
I'm curious why do you need to requeueAfter? Isn't this watching the agent events that would trigger a reconcile when it's intalled?